### PR TITLE
Replace rubocop-govuk with rubocop

### PR DESCRIPTION
**Pull request by despo**
## Context
This PR removes rubocop-govuk and disables a number of cops to match its configuration and avoid having to make a lot of changes across the codebase, so we are able to tackle each cop that we want to re-enable in separate PRs. (Except Metrics/BlockNesting which only resulted in 2 errors that I addressed as part of this PR).

## Changes proposed in this pull request

- Replace rubocop-govuk with rubocop, rubocop-rspec, rubocop-rake and rubocop-rails
- Disable cops to match rubocop-govuk configuration
- Group cop configuration per area they tackle, e.g. rspec.yml, metrics.yml etc

## Link to Trello card
https://trello.com/c/XANNLBur/432-should-we-continue-using-rubocop-govuk
https://trello.com/c/I6JiD7cR/1564-rubocop-cleanup

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
-    - config/default.yml
-    - config/rails.yml
+inherit_from:
+  - .rubocop_todo.yml
+  - ./lib/rubocop/config/rspec.yml
+  - ./lib/rubocop/config/rails.yml
+  - ./lib/rubocop/config/rake.yml
+  - ./lib/rubocop/config/style.yml
+  - ./lib/rubocop/config/layout.yml
+  - ./lib/rubocop/config/metrics.yml
```
Comment (despo) [inline .rubocop.yml:8]: I'm not sure that this is the best place for the configuration, but I do think that everything rubocop related should be grouped. Should I move it all to config/rubocop instead? (along with the custom defined cops?)

Comment (cwrw) [reply .rubocop.yml:8]: Good question. From my perspective it feels ok to separate them, as they serve different purposes. One is the actual config and feels better suited under `config/rubocop`. The cops can still live as now in `lib` as utils which we require.

Im also ok to keep as is though, but not keen on moving things under `config/rubocop` as the cops feel a bit misplaced there

Comment (despo) [reply .rubocop.yml:8]: Ok. I think I'll move the configuration only then to config/rubocop

**Review by cwrw — APPROVED** [review]
Thanks for this @despo! Only comment is that there is now a warning when running rubocop around cops that need setting to true/false: 

```
The following cops were added to RuboCop, but are not configured. Please set Enabled to either `true` or `false` in your `.rubocop.yml` file:
```

Could we mark those as false and add them to our list?

**Comment by despo** [comment]
> Thanks for this @despo! Only comment is that there is now a warning when running rubocop around cops that need setting to true/false: 
> 
> 
> 
> ```
> 
> The following cops were added to RuboCop, but are not configured. Please set Enabled to either `true` or `false` in your `.rubocop.yml` file:
> 
> ```
> 
> 
> 
> Could we mark those as false and add them to our list?

@cwrw sure. 👍

**Pull request merged**